### DISCUSSION TITLE
Expo Notification: addNotificationResponseReceivedListener

### DISCUSSION THREAD

**Discussion by abdelhameedhamdy** [expo/expo#28598] [Archived: Questions]
Hey there,

Wondering, why I cannot set react state variable in the callback response when  addNotificationResponseReceivedListener triggered from killed state, I can only set when backgrounded:

```
/* @info This listener is fired whenever a user taps on or interacts with a notification (works when an app is foregrounded, backgrounded, or killed). */   
responseListener.current = Notifications.addNotificationResponseReceivedListener(response => {     
console.log(response);   
// I want to setSample(response),  it is not working when app is killed, only when app is backgrounded
});
```

**Comment by ItsRD** [comment]
I had the same issue. Whenever the android app is killed, the `addNotificationResponseReceivedListener` is not triggered. In my case this function should be linking the user to a specific screen.

However, after figuring out this was the problem and some debugging later, I noticed there is a last Notification response function. This will return the data from the notification. Maybe more of a workaround than a fix, but it works fine in my case.
```
const notificationResponse = await Notifications.getLastNotificationResponseAsync()
console.log(notificationResponse.notification.request.content)
```
